mkversion.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+set -e
+cd "$(dirname "$0")"
+isdeb=true
+v="$( head -n1 debian/changelog  | sed -r -e 's/^[^ ]+ \((.*)\)( +[^ ]+)+; urgency=.*/\1/' )"
+if [ -z "$DH_GOPKG" ]; then
+    isdeb=false
+    # not building the package; add something to the version
+    if git rev-parse --is-inside-work-tree >&- 2>&-; then
+        # in git; good.
+        v="$( printf "%s+git%s.%04d" "$v" "$( date --utc +%Y%m%d )" "$( git log --since=yesterday --pretty=short | wc -l)" )"
+    else
+        # TODO: add lines for bzr also
+        v="$( printf "%s+unknown%s" "$v" "$( date --utc +%Y%m%d.%H%M%S )" )"
+    fi
+fi
+
+if [ "$1" = "-o" ]; then
```

Comment:
Actually, sorry, that would still not be the usual meaning, since we're writing different contents. Perhaps -w or whatever.. I'll stop the bikeshed now. :-)